File: codility_training/lessons.lesson04.CountingElements.MissingInteger/junilhwang.js
function solution(A) {
  const set = new Set(A)
  for (let i = 1, len = set.size; i <= len; i++) {
    if (!set.has(i)) return i;
  }
  return set.size + 1;
}

console.log(
  solution([1, 3, 6, 4, 1, 2]) === 5,
  solution([1, 2, 3]) === 4,
  solution([-1, -3]) === 1
);

File: programmers/난이도별/level01.완주하지_못한_선수/junilhwang.py
def solution(participant, completion):
    vector = {}
    for v in participant : vector[v] = vector.get(v, 0) + 1
    for v in completion :
        vector[v] -= 1
        if vector[v] == 0 : del vector[v]
    for k, v in vector.items() :
        return k

print(
    solution(['leo', 'kiki', 'eden'], ['eden', 'kiki'] ) == 'leo',
    solution(['marina', 'josipa', 'nikola', 'vinko', 'filipa'], ['josipa', 'filipa', 'marina', 'nikola'] ) == 'vinko',
    solution(['mislav', 'stanko', 'mislav', 'ana'], ['stanko', 'ana', 'mislav']) == 'mislav'
)